rmw_zenoh_cpp/src/detail/graph_cache.hpp

```diff
@@ -23,17 +23,14 @@
 #include <string>
 #include <unordered_map>
 #include <unordered_set>
-#include <vector>
 
 #include "event.hpp"
 #include "liveliness_utils.hpp"
 #include "ordered_map.hpp"
 
 #include "rcutils/allocator.h"
-#include "rcutils/types.h"
```

Comment:
```
#include "rcutils/string_array.h"
```

Reply:
Since clangd doesn't complain about this so I traced the include path. Here's what I found.

`#include "rmw/names_and_types.h"` includes `#include "rcutils/types.h"` that covers `#include "rcutils/types/string_array.h"`

Reply:
I see. I'm unaware of this pattern. We usually trust the compiler in Rust. But when I studied this pattern and tried to apply the tool [include-what-you-use](https://github.com/include-what-you-use/include-what-you-use), I got many warnings. What do you think about introducing the check of include-what-you-use?

[iwyu.txt](https://github.com/user-attachments/files/17827869/iwyu.txt)

Reply:
Furthermore, I think we need to wrap the ROS dependencies with brackets rather than quotes as include-what-you-use suggsted. And this way is also recommended in the ROS community. https://answers.ros.org/question/348742/double-quotes-vs-angled-brackets-for-rosrosh/

Reply:
The suggested header is simply included and move this discussion to https://github.com/ros2/rmw_zenoh/issues/323